app/src/org/commcare/tasks/CleanRawMedia.kt

```diff
@@ -0,0 +1,31 @@
+package org.commcare.tasks
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.commcare.CommCareApplication
+import org.commcare.android.database.user.models.FormRecord
+import org.commcare.utils.CrashUtil
+import org.commcare.utils.StorageUtils.getUnsentCompleteOrSavedFormIdsForCurrentApp
+import java.io.File
+
+class CleanRawMedia(appContext: Context, workerParams: WorkerParameters)
+    : Worker(appContext, workerParams) {
+
+    override fun doWork(): Result {
+        val storage = CommCareApplication.instance().getUserStorage(FormRecord::class.java)
+        val recordsToRemove = getUnsentCompleteOrSavedFormIdsForCurrentApp(storage)
+        for (formId in recordsToRemove) {
+            try {
+                val filePath = storage.getMetaDataFieldForRecord(formId, FormRecord.META_FILE_PATH)
+                File(filePath).parent?.let {
+                    val rawDirPath = "$it/raw"
```

Comment:
Probably best for us to encode this in a static with the other part of the code, to make sure if a change is made that someone knows it affects this cleanup as well.

Reply:
agree though keeping it unchanged for now to avoid another review cycle and  I will implement the suggestion in the `master` version of this PR>